src/pages/index.js

```diff
@@ -68,10 +102,59 @@ editProfileButton.addEventListener("click", () => {
   editProfilePopup.open();
 });
 
+const handleDeleteClick = (card) => {
+  confirmationPopup.open();
+  confirmationPopup.setSubmitCallback(() => {
+    confirmationPopup.setButtonText(true, "Deleting...");
+
+    api
+      .deleteCard(card._cardId)
+      .then(() => {
+        card.delete();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        confirmationPopup.close();
```

Comment:
`        confirmationPopup.close();
`

* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well